packages/lib/gpcircuits/test/virtual-entry.spec.ts

```diff
@@ -0,0 +1,137 @@
+        objectSignerPubkeyAy: padArray(pub_y, paramMaxObjects, pub_y[0])
+      },
+      outputs: {
+        valueHashes: paddedPubHashes,
+        revealedValueHash: paddedPubHashes.map((hash, i) =>
+          paddedRevealedBits[i] === 1n ? hash : BABY_JUB_NEGATIVE_ONE
+        )
+      }
+    };
+  }
+
+  const sampleInput: VirtualEntryModuleInputs = {
+    isValueHashRevealed: 22n,
+    objectSignerPubkeyAx: [
+      13277427435165878497778222415993513565335242147425444199013288855685581939618n,
+      12512819142096328672745574748268841190683864664801826114110182444939815508133n,
+      13277427435165878497778222415993513565335242147425444199013288855685581939618n,
+      13277427435165878497778222415993513565335242147425444199013288855685581939618n,
+      13277427435165878497778222415993513565335242147425444199013288855685581939618n
+    ],
+    objectSignerPubkeyAy: [
+      13622229784656158136036771217484571176836296686641868549125388198837476602820n,
+      13076926918448785155412042385132024413480177434239776354704095450497712564228n,
+      13622229784656158136036771217484571176836296686641868549125388198837476602820n,
+      13622229784656158136036771217484571176836296686641868549125388198837476602820n,
+      13622229784656158136036771217484571176836296686641868549125388198837476602820n
+    ]
+  };
+
+  const sampleOutput: VirtualEntryModuleOutputs = {
+    valueHashes: [
+      8093821485214269328389004542394237209037452657522929891144731833981969398000n,
+      1192297809908180610286326005182698364174835035041287169682595389877239200469n,
+      8093821485214269328389004542394237209037452657522929891144731833981969398000n,
+      8093821485214269328389004542394237209037452657522929891144731833981969398000n,
+      8093821485214269328389004542394237209037452657522929891144731833981969398000n
+    ],
+    revealedValueHash: [
+      21888242871839275222246405745257275088548364400416034343698204186575808495616n,
+      1192297809908180610286326005182698364174835035041287169682595389877239200469n,
+      8093821485214269328389004542394237209037452657522929891144731833981969398000n,
+      21888242871839275222246405745257275088548364400416034343698204186575808495616n,
+      8093821485214269328389004542394237209037452657522929891144731833981969398000n
+    ]
+  };
+
+  it("should accept a sample object", async () => {
+    circuit = await circomkit.WitnessTester("VirtualEntryModule", {
+      file: "virtual-entry",
+      template: "VirtualEntryModule",
+      params: [5]
+    });
+
+    await circuit.expectPass(sampleInput, sampleOutput);
+  });
+
+  it("should accept a dynamic object", async () => {
+    for (const paramMaxObjects of [1, 2, 3, 4, 5]) {
+      circuit = await circomkit.WitnessTester("VirtualEntryModule", {
+        file: "virtual-entry",
+        template: "VirtualEntryModule",
+        params: [paramMaxObjects]
+      });
+      const bitComboGenerator = (
```

Comment:
Can you explain what this does in a comment?  Is this just generating all combinations of `length` bits?  This is another case where the recursive/functional style may be hiding details, vs. for loops and push calls, though it took me some pondering to figure out a clear way to do number-to-bitarray conversion.  How about this:
```ts
    const bitCombos = [];
    for (let n = 0; n < 2**length; n++) {
      bitCombos.push(Array.from(n.toString(2).padStart(length, '0')));
    }
```
Not sure if that's actually much clearer.  The comment is the important thing.